celery_utils/models.py

```diff
@@ -60,3 +67,62 @@ def __str__(self):
             kwargs=self.kwargs,
             resolution='not resolved' if self.datetime_resolved is None else 'resolved'
         )
+
+
+class ChordData(models.Model):
+    """
+    A fairly simply class linking chords together.
+    """
+    completed_results = models.ManyToManyField(TaskMeta, related_name='chorddata_sub_results')
+    serialized_callback = models.TextField()  # A frozen, serialized callback signature
+    callback_result = models.OneToOneField(TaskMeta, related_name='chorddata_callback_result')
+    use_iterator = models.BooleanField(default=False)
```

Comment:
I don't think we actually need this field, we can use `callback_signature['options'].get('use_iterator', False)`